pkg/apiserver/rest/dualwriter.go

```diff
@@ -207,3 +209,75 @@ func SetDualWritingMode(
 
 	return NewDualWriter(currentMode, legacy, storage, reg), nil
 }
+
+var defaultConverter = runtime.UnstructuredConverter(runtime.DefaultUnstructuredConverter)
+
+// Compare asserts on the equality of objects returned from both stores	(object storage and legacy storage)
+func Compare(storageObj, legacyObj runtime.Object) bool {
+	return bytes.Equal(hashUnstructuredObjToCompare(storageObj), hashUnstructuredObjToCompare(legacyObj))
+}
+
+func hashUnstructuredObjToCompare(obj runtime.Object) []byte {
+	cpy := obj.DeepCopyObject()
+	unstObj, err := defaultConverter.ToUnstructured(cpy)
+	if err != nil {
+		return nil
+	}
+	// we don't want to compare meta fields
+	delete(unstObj, "meta")
+
+	ordered := sortUnstructeredMap(unstObj)
+	h := md5.New()
```

Comment:
I'd suggest using https://pkg.go.dev/hash/fnv#New128a , it'll have better performance than md5 and detect changes at least as well.

Reply:
given that we have everythign in memory, I don't see any advantage to even hashing...  just copy:
https://github.com/stretchr/testify/blob/v1.9.0/assert/assertions.go#L1760

then we can skip the sorting etc -- writing JSON from map[string] will always be sorted automatically